src/neo/SmartContract/Native/Oracle/OracleContract.cs

```diff
@@ -0,0 +1,185 @@
+                    int index = (int)(response.Id % (ulong)nodes.Length);
+                    nodes[index].GAS += OracleRequestPrice;
+                }
+            }
+            if (nodes != null)
+            {
+                foreach (var (account, gas) in nodes)
+                {
+                    if (gas.Sign > 0) GAS.Mint(engine, account, gas);
+                }
+            }
+        }
+
+        [ContractMethod(OracleRequestPrice, CallFlags.AllowModifyStates)]
+        private void Request(ApplicationEngine engine, string url, string filter, string callback, StackItem userData, long gasForResponse)
+        {
+            //Check the arguments
+            if (Utility.StrictUTF8.GetByteCount(url) > MaxUrlLength
+                || (filter != null && Utility.StrictUTF8.GetByteCount(filter) > MaxFilterLength)
+                || Utility.StrictUTF8.GetByteCount(callback) > MaxCallbackLength
+                || gasForResponse < 0_10000000)
+                throw new ArgumentException();
+
+            //Mint gas for the response
+            engine.AddGas(gasForResponse);
+            GAS.Mint(engine, Hash, gasForResponse);
+
+            //Increase the request id
+            StorageItem item_id = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_RequestId));
+            ulong id = BitConverter.ToUInt64(item_id.Value) + 1;
+            item_id.Value = BitConverter.GetBytes(id);
+
+            //Put the request to storage
+            if (engine.Snapshot.Contracts.TryGet(engine.CallingScriptHash) is null)
+                throw new InvalidOperationException();
+            engine.Snapshot.Storages.Add(CreateStorageKey(Prefix_Request).Add(item_id.Value), new StorageItem(new OracleRequest
+            {
+                OriginalTxid = GetOriginalTxid(engine),
+                GasForResponse = gasForResponse,
+                Url = url,
+                Filter = filter,
+                CallbackContract = engine.CallingScriptHash,
+                CallbackMethod = callback,
+                UserData = BinarySerializer.Serialize(userData, MaxUserDataLength)
+            }));
+
+            //Add the id to the IdList
+            engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_IdList).Add(GetUrlHash(url)), () => new StorageItem(new IdList())).GetInteroperable<IdList>().Add(id);
+        }
+
+        [ContractMethod(0, CallFlags.AllowModifyStates)]
+        private void SetOracleNodes(ApplicationEngine engine, ECPoint[] nodes)
+        {
+            if (nodes.Length == 0) throw new ArgumentException();
+            if (!CheckCommittee(engine)) throw new InvalidOperationException();
+            NodeList list = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_NodeList)).GetInteroperable<NodeList>();
+            list.Clear();
+            list.AddRange(nodes);
+        }
+
+        [ContractMethod(0_01000000, CallFlags.AllowStates)]
+        private bool Verify(ApplicationEngine engine)
```

Comment:
The `verify` of native contract, can not be called directly in `Helper.VerifyWitness`  and `Wallet.CalculateNetworkFee`

https://github.com/neo-project/neo/blob/8879886f8fb694cd2f274cca7675442b3996d812/src/neo/SmartContract/Helper.cs#L177-L180

https://github.com/neo-project/neo/blob/8879886f8fb694cd2f274cca7675442b3996d812/src/neo/Wallets/Wallet.cs#L382-L388

Reply:
Igore it, we can construct the witnesses of response tx directly.